core/src/mindustry/world/Block.java

```diff
@@ -430,7 +430,7 @@ public void setBars(){
             boolean buffered = cons.buffered;
             float capacity = cons.capacity;
 
-            bars.add("power", entity -> new Bar(() -> buffered ? Core.bundle.format("bar.poweramount", Float.isNaN(entity.power.status * capacity) ? "<ERROR>" : (int)(entity.power.status * capacity)) :
+            bars.add("power", entity -> new Bar(() -> buffered ? Core.bundle.format("bar.poweramount", Float.isNaN(entity.power.status * capacity) ? "<ERROR>" : entity.power.status * capacity > 1000 ? UI.formatAmount((long)(entity.power.status * capacity)) : (int)(entity.power.status * capacity)) :
```

Comment:
The 1k check should not be necessary, UI.formatAmount doesn't change the number when it's <1k anyway.